includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -57,14 +65,32 @@ public static function factory(
 			$repo->newEditEntityFactory(),
 			$editRequestParser,
 			$reconciliationService,
-			$simplePutStrategy
+			$simplePutStrategy,
+			// @TODO Inject this, when there is a good way to do that
+			RequestContext::getMain()->getUser()
 		);
 	}
 
 	public function run() {
-		// Get the request
+		// Rest Validator returns null if submitted as form
+		$requestBody = $this->getValidatedBody();
+		if ( $requestBody === null ) {
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-reconcile-json' ),
+				415
+			);
+		}
+
+		if ( !$this->user->isRegistered() || !$this->user->matchEditToken( $requestBody['token'] ) ) {
```

Comment:
If we’re checking the edit token here anyways (which I think is a good idea, because it happens before we start handing out new item IDs), then we could probably continue to pass `false` into `attemptSave()`? Or is it useful to do the token check twice?

Reply:
Feels a bit backwards not using the token when saving if it was included for this purpose?

Reply:
Fair enough ^^